src/components/linter.ts

```diff
@@ -1,248 +1,50 @@
 import * as vscode from 'vscode'
-import * as path from 'path'
-import * as fs from 'fs'
-import * as os from 'os'
-import {ChildProcessWithoutNullStreams, spawn} from 'child_process'
-import {EOL} from 'os'
 
 import type {Extension} from '../main'
+import { ChkTeX } from './linterlib/chktex'
+import { LaCheck } from './linterlib/lacheck'
 
 export class Linter {
-    private readonly extension: Extension
+    private readonly chktex: ChkTeX
+    private readonly lacheck: LaCheck
     private linterTimeout?: NodeJS.Timer
-    private readonly currentProcesses = Object.create(null) as { [linterId: string]: ChildProcessWithoutNullStreams }
 
-    constructor(extension: Extension) {
-        this.extension = extension
+    constructor(private readonly extension: Extension) {
+        this.chktex = new ChkTeX(extension)
+        this.lacheck = new LaCheck(extension)
     }
 
-    private get rcPath() {
-        let rcPath: string
-        // 0. root file folder
-        const root = this.extension.manager.rootFile
-        if (root) {
-            rcPath = path.resolve(path.dirname(root), './.chktexrc')
+    private getLinters(scope?: vscode.ConfigurationScope): (ChkTeX | LaCheck)[] {
```

Comment:
Why not defining a linter interface that `chkTeX` and `LaCheck` would implement? Something like
```ts
interface ILinter {
  readonly linterDiagnostics: vscode.DiagnosticCollection,
  readonly #linterName: string,
  async lintRootFile(),
  async lintFile(document: vscode.TextDocument),
  parseLog(log: string, filePath?: string)
}
```

This would allow `getLinters` to return a `ILinter[]` and more importantly it would clarify what has to be implemented to add a new linter.